### PR TITLE
[gallery lib playground] feat: added new feature overlayDesign

### PR DESCRIPTION
**Pull request by yuvalab**

### CODE REVIEW

File: packages/gallery/src/components/item/itemHover.js [hunk 1]
```diff
@@ -54,17 +54,107 @@ export default class ItemHover extends GalleryComponent {
...
+      overlayPosition === GALLERY_CONSTS.overlayPositions.RIGHT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.CENTERED_HORIZONTALLY;
+    if (isHorizontal) {
+      return this.calculateHorizontalOverlayStyles();
+    }
+    return this.calculateVerticalOverlayStyles();
+  }
+
+  calculateVerticalOverlayStyles() {
+    const { styleParams, imageDimensions } = this.props;
```
Comment (nirnaor) [inline packages/gallery/src/components/item/itemHover.js]: Remove calculations outside of the component. All these calculations are dependant on a few props (from the `styleParams`) therefor should not be performed inside a component that renders

File: packages/gallery/src/components/item/itemHover.js [hunk 2]
```diff
@@ -54,17 +54,107 @@ export default class ItemHover extends GalleryComponent {
...
+    let style = { maxHeight: height, maxWidth: width };
+    let overlayHeight =
+      (overlaySizeType === 'PERCENT'
+        ? height * (overlaySize / 100)
+        : overlaySize) -
+      2 * overlayPadding;
+    let overlayWidth = width - 2 * overlayPadding;
+    let marginRight = overlayPadding;
+    let marginBottom = overlayPadding;
+    marginTop += overlayPadding;
```
Comment (nirnaor) [inline packages/gallery/src/components/item/itemHover.js]: New code should be written [without mutations](https://github.com/immutable-js/immutable-js/#introduction).  This is out of scope for our meeting but still important

File: packages/playground/src/constants/settings.js [hunk 3]
```diff
@@ -116,7 +116,10 @@ export const stylesBySection = {
     'arrowsVerticalPosition',
 
     'overlayBackground',
-
+    'overlayPosition',
```
Comment (nirnaor) [inline packages/playground/src/constants/settings.js:121]: Why can't we iterate over something inside lib instead of copying the new keys?

**Review by nirnaor — CHANGES_REQUESTED** [review]

File: packages/gallery/src/components/item/itemHover.js [hunk 4]
```diff
@@ -54,22 +54,109 @@ export default class ItemHover extends GalleryComponent {
...
+    } = styleParams;
+    let { width, height } = imageDimensions;
+    let style = {};
+    let overlayHeight, overlayWidth, overlaySize;
+    const margin = overlayPadding;
+    const isHorizontal =
+      overlayPosition === GALLERY_CONSTS.overlayPositions.LEFT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.RIGHT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.CENTERED_HORIZONTALLY;
+    console.log(width);
```
Comment (nirnaor) [inline packages/gallery/src/components/item/itemHover.js]: remove

File: packages/gallery/src/components/item/itemHover.js [hunk 5]
```diff
@@ -54,22 +54,109 @@ export default class ItemHover extends GalleryComponent {
...
+    const { styleParams, imageDimensions } = this.props;
+    let {
+      overlayPosition,
+      overlaySize: requiredOverlaySize,
+      overlaySizeType,
+      overlayPadding,
+    } = styleParams;
+    let { width, height } = imageDimensions;
+    let style = {};
+    let overlayHeight, overlayWidth, overlaySize;
```
Comment (nirnaor) [inline packages/gallery/src/components/item/itemHover.js]: can you try and make the entire method with `const` only? I want us to stop adding mutations to the code

File: packages/gallery/src/components/item/itemHover.js [hunk 6]
```diff
@@ -54,22 +54,109 @@ export default class ItemHover extends GalleryComponent {
...
+    return style;
+  }
+
+  getOverlayPositionByFlex() {
+    const { styleParams } = this.props;
+    const { overlayPosition } = styleParams;
+    let style = {
+      width: '100%',
+      height: '100%',
+      display: 'flex',
```
Comment (nirnaor) [inline packages/gallery/src/components/item/itemHover.js]: Can you try without mutations?

File: packages/gallery/tests/e2e/styleParams/overlayDesign.e2e.spec.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      overlayPosition: GALLERY_CONSTS.overlayPositions.CENTERED_HORIZONTALLY,
+      overlaySize: 50,
+      overlaySizeType: 'PERCENT',
+      overlayPadding: 30,
+      hoveringBehaviour: GALLERY_CONSTS.infoBehaviourOnHover.NO_CHANGE,
+    });
+    await driver.waitFor.hookToBeVisible('item-container');
+    await driver.waitFor.timer(200);
+    const page = await driver.grab.elemScreenshot('.pro-gallery');
+    expect(page).toMatchImageSnapshot();
```
Comment (nirnaor) [inline packages/gallery/tests/e2e/styleParams/overlayDesign.e2e.spec.js:31]: This code is duplicated :-( 
Makes it really hard to understand what is the difference between the tests.

File: packages/gallery/tests/styleParams/overlayPosition.spec.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    Object.assign(initialProps.styles, {
+      galleryLayout: GALLERY_CONSTS.layout.GRID,
+      overlayPosition: GALLERY_CONSTS.overlayPositions.TOP,
+      overlaySize: 50,
+      overlaySizeType: 'PIXEL',
+      overlayPadding: 30,
+    });
+    driver.mount.proGallery(initialProps);
+    await driver.update();
+    const overlaySizeImage = driver.find.selector('.gallery-item-hover').at(0);
```
Comment (nirnaor) [inline packages/gallery/tests/styleParams/overlayPosition.spec.js:29]: can you reuse the logic ?

File: packages/gallery/tests/styleParams/overlayPosition.spec.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    console.log(overlaySizeImage.props().style);
```
Comment (nirnaor) [inline packages/gallery/tests/styleParams/overlayPosition.spec.js]: remove debugging stuff

File: packages/gallery/tests/styleParams/overlaySize.spec.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  it('should set the correct overlaySize of "50" - position top ', async () => {
+    Object.assign(initialProps.styles, {
+      galleryLayout: GALLERY_CONSTS.layout.GRID,
+      overlaySize: 50,
+      overlayPosition: GALLERY_CONSTS.overlayPositions.TOP,
+      overlayPadding: 30,
+    });
+    driver.mount.proGallery(initialProps);
```
Comment (nirnaor) [inline packages/gallery/tests/styleParams/overlaySize.spec.js]: duplicated

File: packages/gallery/tests/styleParams/overlaySizeType.spec.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    const { width } = overlaySizeImage.props().style;
+    expect(width).to.eq(20);
+    driver.detach.proGallery();
+  });
+
+  it('should set the correct overlaySize according to size type- PERCENT', async () => {
+    Object.assign(initialProps.styles, {
+      galleryLayout: GALLERY_CONSTS.layout.GRID,
+      overlayPosition: GALLERY_CONSTS.overlayPositions.LEFT,
+      overlaySizeType: 'PERCENT',
```
Comment (nirnaor) [inline packages/gallery/tests/styleParams/overlaySizeType.spec.js:39]: duplicated

**Review by nirnaor — CHANGES_REQUESTED** [review]
let me know if you need help with the `const` thing.

File: packages/gallery/src/components/item/itemHover.js (hunk 4 above)
Comment (IzaacAyelin) [inline packages/gallery/src/components/item/itemHover.js]: delete log

**Pull request merged**